### ISSUE TITLE
some details about similarity subnet

### ISSUE THREAD

**Issue by Zebrocode** [msracver/Deep-Exemplar-based-Colorization#14]
I am trying to train the grey-VGG-19 in the paper ,but using pytorch. Could you please tell me some details about training it?
1.which dataset do you use to trian the VGG-19?
2.How did you transfer the dataset from RGB to Lab ?
3.The bidirectional map function just need to be used after the VGG-19 net ,am I right?
4.which Loss function/optimizer  did you use in VGG-19 when training VGG-19?
sorry to both you...

**Comment by hmmlillian** [comment]
@Zebrocode 
1. We use the original ImageNet to train;
2. The coversion from RGB to Lab:
``img_lab = cv2.cvtColor(img, cv2.COLOR_BGR2LAB)
  l, a, b = cv2.split(img_lab)
  img = cv2.merge((l, l, l))```
3. Yes, the bidirectional map is computed with the features from pre-trained network;
4. We trained it following [this](https://github.com/cvjena/cnn-models/tree/master/VGG19_cvgj) with a SoftmaxWithLossLayer in Caffe.

**Comment by Zebrocode** [comment]
Hello,it's me again...I came to the Colorization Sub-Network,but it confused me at the start, how could i
use simT<-->R together with T_L and T_ab. How do i deal with the three inputs to make it as one to feed the U-net you used in the paper. Thank you for your help!

**Comment by hmmlillian** [comment]
@Zebrocode 
 We concatenate all the inputs in one tensor (refer to  `Deep-Exemplar-based-Colorization/colorization_subnet/test.py`):
`colornet_input = torch.cat((im_l, warped_ba[:, 1:, ...], err_ba, err_ab), dim=1)`
and input to the network.
